src/main/kotlin/uk/gov/justice/digital/hmpps/notificationsalertsvsip/service/EmailSenderService.kt

```diff
@@ -0,0 +1,108 @@
+package uk.gov.justice.digital.hmpps.notificationsalertsvsip.service
+
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.notificationsalertsvsip.config.TemplatesConfig
+import uk.gov.justice.digital.hmpps.notificationsalertsvsip.dto.SendEmailNotificationDto
+import uk.gov.justice.digital.hmpps.notificationsalertsvsip.dto.personalisations.PrisonerVisitorPersonalisationDto
+import uk.gov.justice.digital.hmpps.notificationsalertsvsip.dto.prisoner.contact.registry.PrisonerContactRegistryContactDto
+import uk.gov.justice.digital.hmpps.notificationsalertsvsip.dto.visit.scheduler.VisitDto
+import uk.gov.justice.digital.hmpps.notificationsalertsvsip.enums.EmailTemplateNames
+import uk.gov.justice.digital.hmpps.notificationsalertsvsip.enums.VisitEventType
+import uk.gov.justice.digital.hmpps.notificationsalertsvsip.enums.visit.scheduler.VisitRestriction
+import uk.gov.justice.digital.hmpps.notificationsalertsvsip.utils.DateUtils
+import uk.gov.service.notify.NotificationClient
+import java.time.LocalDate
+import java.time.temporal.ChronoUnit
+
+@Service
+class EmailSenderService(
+  @Value("\${notify.email.enabled:}") private val enabled: Boolean,
+  val notificationClient: NotificationClient,
+  val prisonRegisterService: PrisonRegisterService,
+  val prisonerSearchService: PrisonerSearchService,
+  val prisonerContactRegistryService: PrisonerContactRegistryService,
+  val templatesConfig: TemplatesConfig,
+  val dateUtils: DateUtils,
+) {
+  private companion object {
+    val LOG: Logger = LoggerFactory.getLogger(this::class.java)
+  }
+
+  fun sendEmail(visit: VisitDto, visitEventType: VisitEventType) {
+    if (enabled) {
+      val sendEmailNotificationDto = when (visitEventType) {
+        VisitEventType.BOOKED -> {
+          handleBookedEvent(visit)
+        }
+
+        else -> {
+          LOG.error("visit event type $visitEventType is unsupported")
+          return Unit
+        }
+      }
+
+      notificationClient.sendEmail(
+        templatesConfig.emailTemplates[sendEmailNotificationDto.templateName.name],
+        visit.visitContact.email,
+        sendEmailNotificationDto.templateVars,
+        visit.reference,
+      )
+    } else {
+      LOG.info("Sending Email has been disabled.")
+    }
+  }
+
+  private fun handleBookedEvent(visit: VisitDto): SendEmailNotificationDto {
+    val templateVars = mutableMapOf(
+      "ref number" to visit.reference,
+      "prison" to prisonRegisterService.getPrisonName(visit.prisonCode),
+      "time" to dateUtils.getFormattedTime(visit.startTimestamp.toLocalTime()),
+      "end time" to dateUtils.getFormattedTime(visit.endTimestamp.toLocalTime()),
+      "arrival time" to "45",
```

Comment:
should we get this from a service with a hardcoded value of 45 later update the service to get it from the visit-scheduler?

Reply:
For now, it's unlikely to be changed for a long time. I'm happy to leave it for now 😎